### DISCUSSION TITLE
String validation in the query parameters to accept the values based on predefined values

### DISCUSSION THREAD

**Discussion by saeedesmaili** [fastapi/fastapi#7293] [Questions] [answered]
### First check

* [x] I added a very descriptive title to this issue.
* [x] I used the GitHub search to find a similar issue and didn't find it.
* [x] I searched the FastAPI documentation, with the integrated search.
* [x] I already searched in Google "How to X in FastAPI" and didn't find any information.
* [x] I already read and followed all the tutorials in the docs and didn't find an answer.
* [x] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
* [x] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
* [x] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).
* [x] After submitting this, I commit to one of:
    * Read open issues with questions until I find 2 issues where I can help someone and add a comment to help there.
    * I already hit the "watch" button in this repository to receive notifications and I commit to help at least 2 people that ask questions in the future.
    * Implement a Pull Request for a confirmed bug.

### Description

I want to use query parameters to pass a string value to the GET request, but it can't be any string and should be one of the predefined ones. Assume that we want to pass one of the `rating`, `published_at`, and `alphabetical` using the query parameter `sorted_by`, so if anything else rather than these three has been sent to the function, we don't want to accept it. 

I found [Query Parameters and String Validation](https://fastapi.tiangolo.com/tutorial/query-params-str-validations/) section of the documentation that we can specify for example the `max_length' of the string, but couldn't want anything related to my problem.

**Comment by SirTelemak** [comment, marked as answer]
You should use Enums for such cases:
```python3
router = FastAPI()

class Values(Enum):
    PUBLISHED_AT = 'published_at'
    RATING = 'rating'

@router.get("/pydanticmodel")
def test(value: Values):
    ...
```
![image](https://user-images.githubusercontent.com/9435877/93077334-5382e700-f691-11ea-9b05-bd5a23e343a9.png)

**Comment by tiangolo** [comment]
Thanks for the help here @SirTelemak! :clap: :bow: 

Thanks for reporting back and closing the issue @saeedesmaili  :+1:
